test/Microsoft.Health.Fhir.Shared.Tests.E2E/Rest/EverythingOperationTestFixture.cs

```diff
@@ -90,6 +101,35 @@ protected override async Task OnInitializedAsync()
             ObservationOfNonExistentPatient = await TestFhirClient.CreateAsync(observationToCreate);
 
             await TestFhirClient.DeleteAsync(NonExistentPatient);
+
+            // Test case #3
+            // Create patients with different types of links
+            patientToCreate = Samples.GetJsonSample<Patient>("PatientWithMinimalData");
+            patientToCreate.Id = "patient-reference";
+            PatientReferencedByLink = await TestFhirClient.CreateAsync(patientToCreate);
+
+            PatientWithSeeAlsoLink = await CreatePatientWithLink(Patient.LinkType.Seealso, PatientReferencedByLink);
```

Comment:
>PatientWithSeeAlsoLink [](http://example.com/codeflow?start=12&length=22)

What would happened if I have loop?
A pointing to B, B pointing to A?

Reply:
We only process `seealso` links one layer deep. So, if patient A has a `seealso` link to patient B, we will:
1. run $everything on patient A, storing its `seealso` links
2. Check if there are any `seealso` links left to process --> we find that patient B needs to be processed
2. run $everything on patient B , but do not store its `seealso` links

Reply:
A few more loop scenarios that I covered:

- If patient A has a link to itself, we do not add the link as something that needs to be processed
- If patient A has two links to patient B, we only add one of them to the list of links to be processed

Please let me know if there is anything I missed!

Reply:
I know other people are smarter than me, but I couldn't figure that out from glancing over the code. Would it worth to put some comments about how patient everything works with these links somewhere in a code?

Reply:
Added comments in a8f5b52.